common/djangoapps/entitlements/models.py

```diff
@@ -24,3 +24,25 @@ class CourseEntitlement(TimeStampedModel):
         help_text='The current Course enrollment for this entitlement. If NULL the Learner has not enrolled.'
     )
     order_number = models.CharField(max_length=128, null=True)
+
+    @classmethod
+    def get_active_user_course_entitlements(cls, user, course_uuid):
+        """
+        Returns all the available sessions for a given course.
+        """
+
+        try:
+            entitlement = cls.objects.get(
+                user=user,
+                course_uuid=course_uuid,
+            )
```

Comment:
Can a user have more than one entitlement for a course? (Maybe both 'credit' and 'verified'?

Reply:
This all relies on the assumption a course will only ever have one paid mode.  Also Credit is not a great example because of how that is applied. 

This question makes sense when thinking about 'verified' and 'professional' though.

I will double check, but I believe a Course Entitlement and its runs all should have the same modes.

Reply:
This method is dead code. Deleting.